pose_estimator/src/pose_estimator.cpp

```diff
@@ -32,361 +32,321 @@
-          Diagnostic diagnostic{Diagnostic::OK};
-          if (std::abs(rpy.z) > maxYawAng) {
-            diagnostic.status(Diagnostic::ERROR)
-                .description("Yaw Angle - NOT OK -> |%f rad| > %f rad", rpy.z, maxYawAng)
-                .code(health_monitor::ReportFault::POSE_HEADING_THRESHOLD_REACHED);
-          } else {
-            diagnostic.description("Yaw Angle - OK (%f)", rpy.z);
-          }
-          return diagnostic;
-        });
-
-    diagnosticsUpdater.add(orientationYawCheck);
-
-    // Initialize corrected data header
-    m_correctedData.header.frame_id = "corrected_data";
+  else
+  {
+    depth = pressure / GRAVITY / FRESHWATER_DENSITY;
   }
+  return depth;
+}
 
-  ~PoseEstimatorNode() { stop(); }
-
-  int start() {
-    stop();
-    running = true;
-    // Start data publishing timer
-    ROS_INFO("Starting data publishing timer");
-    m_timerPub = node_handle.createTimer(ros::Duration(1.0 / pub_rate),
-                                         boost::bind(&PoseEstimatorNode::publishData, this));
-    return 0;
-  }
+int PoseEstimatorNode::start()
+{
+  stop();
+  running = true;
+  // Start data publishing timer
+  ROS_INFO("Starting data publishing timer");
+  m_timerPub = node_handle.createTimer(ros::Duration(1.0 / pubRate),
+                                       boost::bind(&PoseEstimatorNode::publishData, this));
+  return 0;
+}
 
-  int stop() {
-    if (running) {
-      running = false;
-      m_timerPub.stop();
-    }
-    return 0;
+int PoseEstimatorNode::stop()
+{
+  if (running)
+  {
+    running = false;
+    m_timerPub.stop();
   }
+  return 0;
+}
 
-  bool spin() {
-    while (!ros::isShuttingDown()) {
-      if (start() == 0) {
-        while (node_handle.ok()) {
-          ros::spin();
-        }
+bool PoseEstimatorNode::spin()
+{
+  while (!ros::isShuttingDown())
+  {
+    if (start() == 0)
+    {
+      while (node_handle.ok())
+      {
+        ros::spin();
       }
     }
-    stop();
-    return true;
-  }
-
-  void ahrsDataCallback(const sensor_msgs::Imu data) {
-    boost::mutex::scoped_lock lock(m_mutDataLock);
-    tf::Quaternion quat(data.orientation.x, data.orientation.y, data.orientation.z,
-                        data.orientation.w);
-    tf::Matrix3x3 m(quat);
-    double roll, pitch, yaw;
-    m.getRPY(roll, pitch, yaw);
-    cur_rpy_ang[CorrectedData::ROLL] = roll * -1.0;
-    cur_rpy_ang[CorrectedData::PITCH] = pitch * -1.0;
-    cur_rpy_ang[CorrectedData::YAW] = yaw;
-    // ROS_INFO("RPY: [%f,%f,%f]",roll*180.0/M_PI,pitch*180.0/M_PI,yaw*180.0/M_PI);
-    ahrs_ok = true;
   }
+  stop();
+  return true;
+}
 
-  void gpsDataCallback(const sensor_msgs::NavSatFix data) {
-    if (data.status.status >= 0) {
-      boost::mutex::scoped_lock lock(m_mutDataLock);
-      cur_latitude = data.latitude;
-      cur_longitude = data.longitude;
-      cur_gps_time = data.header.stamp.toSec();
-    }
-  }
+void PoseEstimatorNode::ahrsDataCallback(const sensor_msgs::Imu data)
```

Comment:
@patapufate nit: make it a const ref
```suggestion
void PoseEstimatorNode::ahrsDataCallback(const sensor_msgs::Imu& data)
```

Reply:
fixed